CHIA-1472 Augment get_mirror_info to accept both types of programs

AmineKhaldi opened the pull request:
### Purpose:

This allows us to use serialized programs without the need to convert them into programs.

### Current Behavior:

We would need to convert serialized programs to programs before using the function.

### New Behavior:

We no longer need to convert serialized programs into programs before using the function.

arvidn commented on chia/wallet/db_wallet/db_wallet_puzzles.py:
```diff
@@ -94,8 +94,10 @@ def create_mirror_puzzle() -> Program:
+def get_mirror_info(
+    parent_puzzle: Union[Program, SerializedProgram], parent_solution: Union[Program, SerializedProgram]
+) -> Tuple[bytes32, List[bytes]]:
+    _, conditions = parent_puzzle.run_with_cost(INFINITE_COST, parent_solution)
```
are you sure it's OK to pass a `SerializedProgram` to `Program.run()`? Will it deserialize it correctly and turn it into a `SExp`? I'm not sure it does

AmineKhaldi replied:
We call this with the same type for puzzle and solution, so I added an assert to signal that intent.

Starttoaster replied:
Closing thread, taking Arvid's approval as implicit conversation resolution.

AmineKhaldi committed:
Augment get_mirror_info to accept both types of programs.

This allows us to use serialized programs without the need to convert them into programs.
